cdap-proto/src/main/java/co/cask/cdap/proto/id/ArtifactId.java

```diff
@@ -33,6 +33,13 @@
 
   public ArtifactId(String namespace, String artifact, String version) {
     super(namespace, EntityType.ARTIFACT);
+    if (artifact == null) {
```

Comment:
Why for ApplicationId we explicitly disallow empty string, but here we only check for null?
I don't think the `.isEmpty()` check is needed in ApplicationId, because `ensureValidId` would check that anyways.